lib/explorer/series.ex

```diff
@@ -1369,6 +1369,55 @@ defmodule Explorer.Series do
   @spec at(series :: Series.t(), idx :: integer()) :: any()
   def at(series, idx), do: fetch!(series, idx)
 
+  @doc """
+  Returns a string series with all values ​​concatenated.
+
+  ## Examples
+
+      iex> s1 = Explorer.Series.from_list(["a", "b", "c"])
+      iex> s2 = Explorer.Series.from_list(["d", "e", "f"])
+      iex> s3 = Explorer.Series.from_list(["g", "h", "i"])
+      iex> Explorer.Series.format([s1, s2, s3])
+      #Explorer.Series<
+        Polars[3]
+        string ["adg", "beh", "cfi"]
+      >
+
+      iex> s1 = Explorer.Series.from_list(["a", "b", "c", "d"])
+      iex> s2 = Explorer.Series.from_list([1, 2, 3, 4])
+      iex> s3 = Explorer.Series.from_list([1.5, :nan, :infinity, :neg_infinity])
+      iex> Explorer.Series.format([s1, "/", s2, "/", s3])
+      #Explorer.Series<
+        Polars[4]
+        string ["a/1/1.5", "b/2/NaN", "c/3/inf", "d/4/-inf"]
+      >
+
+      iex> s1 = Explorer.Series.from_list([<<1>>, <<239, 191, 19>>], dtype: :binary)
+      iex> s2 = Explorer.Series.from_list([<<3>>, <<4>>], dtype: :binary)
+      iex> Explorer.Series.format([s1, s2])
+      ** (RuntimeError) External error: invalid utf-8 sequence
+  """
+  @doc type: :shape
+  @spec format([
+          Series.t() | number() | Date.t() | Time.t() | NaiveDateTime.t() | boolean() | String.t()
+        ]) :: Series.t()
+  def format([%Series{} = h | t] = _list), do: Enum.reduce(t, h, &format_reducer/2)
+
+  def format([h | t] = _list),
+    do: Enum.reduce(t, from_list([h], dtype: :string), &format_reducer/2)
```

Comment:
Instead of formatting one by one, I think you can pass the whole list at once. The implementation will be something like:

```elixir
def format([_ | _] = list) do
  Shared.apply_series_impl(:format, cast_to_string(list))
end
```

where cast_to_string will call `from_list` on a string and cast non-string types. Then you will do the reduce in Rust (or in the Polars backend!). In other words, the callback will be `format/1`.

Reply:
Yes, it makes sense! (https://github.com/elixir-nx/explorer/pull/520/commits/42c429f6e8b07483df881afdb87f807b84c4e7e0)